### ISSUE TITLE
ValidationError message contains newlines

### ISSUE THREAD

**Issue by roysmith** [hmarr/mongoengine#539]
If you get a validation error, str() of the ValidationError instance raised contains newlines:

Errors encountered validating document:\ndasherized_name: Field is required ("dasherized_name")

This really complicates logging via syslog, since the message gets broken up into two syslog entries (which may not even be adjacent to each other in a busy system).  I would leave out the entire "Errors encountered validating document" part (that's obvious from the name of the exception).  Just use the rest of it as the exception message (with no newline).

**Comment by rozza** [comment]
Sounds fair - how are you logging to syslog?  I think people output the errors in forms as well - so need not break their code.

**Comment by roysmith** [comment]
We use the standard python logging module.  The logging config looks something like:

```
    'handlers': {
        [...]
        'syslog': {
            'level': 'DEBUG',
            'class': 'logging.StreamHandler',
            'formatter': 'console-verbose',
            'class': 'logging.handlers.SysLogHandler',
            'facility': logging.handlers.SysLogHandler.LOG_LOCAL5,
            'address': '/dev/log',
            'formatter': 'verbose',
        },
```

Do people really depend on how the exception message is formatted?  They shouldn't.

In general, exception messages should be pretty terse.  The "Errors encountered validating document" part doesn't add anything; it's redundant with the fact that this is a ValidationError.  And certainly there's no reason to say "dasherized_name" twice in "dasherized_name: Field is required ("dasherized_name")".

The problem with verbose messages like this is they lock the developer into being unable to do anything useful with the message other than display it verbatim to the user, which in turn forces the UI to do things the application developer may not want.  In my case, all I want is to dump the message into syslog.  But, somebody else may want to (as you say), display it on a form, make it pop up as a mouse-over in a browser, convert it to speech, translate it to a different language, etc.  On a mobile device, screen real-estate is in short supply, so displaying a two-line error message may be impossible.

On the other hand, if you give somebody just the short version, it's easy for them to adorn it whatever way makes sense for their application.

What might be even better is a richer set of ValidationExceptions.  I could see FieldRequired being a subclass of ValidationError, and including the field name as an attribute of the exception object.  Then, an application could do something like:

```
try:
    document.save()
except FieldRequired as ex:
    print ("You need to include the %s field" % ex.field_name)
```

If they didn't want to do something this specific, they could just print str(ex) and call it a day.  But it would be up to the developer to decide what was the right treatment.

**Comment by rozza** [comment]
Sure - I wasn't aware of people logging them thats all - perhaps adding a logger with verboseness would be better.

I thought the exceptions were mainly for used in development.  Also there is a dict that can be used for determining errors against fields when outputting forms etc..

**Comment by rozza** [comment]
will be fixed in 0.6.15
